weave/tests/test_client_trace.py

```diff
@@ -766,3 +766,66 @@ def test_dataset_row_type(client):
         d = weave.Dataset(rows=[{"a": 1}, "a", "b"])
     with pytest.raises(ValidationError):
         d = weave.Dataset(rows=[{"a": 1}, {}])
+
+
+def test_unknown_input_and_output_types(client):
+    class MyUnserializableClassA:
+        a_val: int
+
+        def __init__(self, a_val) -> None:
+            self.a_val = a_val
+
+    class MyUnserializableClassB:
+        b_val: int
+
+        def __init__(self, b_val) -> None:
+            self.b_val = b_val
+
+    @weave.op()
+    def op_with_unknown_types(
+        a: MyUnserializableClassA, b: int
+    ) -> MyUnserializableClassB:
+        return MyUnserializableClassB(a.a_val + b)
+
+    a = MyUnserializableClassA(3)
+    res = op_with_unknown_types(a, 0.14)
```

Comment:
Did you mean to use `float` instead of `int` in your type annotations?

Reply:
👍